### PR TITLE
feat(Button): add icon-only button support

### LINKED ISSUE #2490: Expose icon-only button in Carbon React documentation
Related: https://github.com/IBM/carbon-components/pull/2287, https://github.com/carbon-design-system/carbon-components/pull/2366

### PR DESCRIPTION
**Pull request by emyarod**
Closes #2490

This PR adds support for icon-only buttons in the React library

Refs #2287, #2366, #3115

#### Changelog

**New**

- Icon only button support and docs for React library

**Changed**

- update vanilla icon-only button example

**Removed**

- unused CSS class in vanilla docs (button, overflow menu, icon tooltip)

#### Testing / Reviewing

Ensure that icon-only buttons appear and function as expected

### CODE REVIEW

File: packages/components/src/components/button/button.hbs [hunk 1]
```diff
@@ -28,27 +28,27 @@
...
+      placeholder="Optional placeholder text">
+  </div>
+  <div class="example-button-wrapper">
+    <button class=" {{@root.prefix}}--btn {{@root.prefix}}--btn--{{variant}}{{#if small}}
       {{@root.prefix}}--btn--sm{{/if}}{{#if field}} {{@root.prefix}}--btn--field{{/if}}" {{#if danger}}
-        aria-label="danger" {{/if}} type="button">
-        Button
-      </button>
-    </div>
+      aria-label="danger" {{/if}} type="button">
```
Comment (elizabethsjudd) [inline packages/components/src/components/button/button.hbs:49]: FYI, if you add an `aria-label` to a button the SR doesn't actually read the contents of them button so the user would never hear `Button`.  I had a chat with @snidersd about this and she said that for danger there really isn't a way to denote this variant generically to the user but we be following best practices for content design and making sure the label of the button is clear to the user.

**Review by elizabethsjudd — COMMENTED** [review]
Thanks for taking this work over fo the React demos!!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod for doing this!

**Review by tw15egan — APPROVED** [review]
Nice work 👍

**Review by aagonzales — APPROVED** [review]
Looks good! 🎉

File: packages/components/src/components/button/button.hbs (hunk 1 above)
Comment (emyarod) [reply packages/components/src/components/button/button.hbs:49]: thanks for the info, this was just carried over from our existing button examples. should we open a new PR to remove that `aria-label` in all of the examples?

Comment (elizabethsjudd) [reply packages/components/src/components/button/button.hbs:49]: I agree that it's beyond the scope of this PR and should be moved to a separate issue.

**Pull request merged**